managers/docker-manager/DockerService.js

```diff
@@ -0,0 +1,801 @@
+  //     return `Up for ${utils.getTimeAgo(state.StartedAt, true)}${state.Paused ? ' (Paused)' : ''}`;
+  //   }
+  //   if (state.ExitCode > 0) {
+  //     return `Exited (${state.ExitCode}) ${utils.getTimeAgo(state.FinishedAt)}`;
+  //   }
+  //   return 'Stopped';
+  // }
+
+  getEnvironment() {
+    return _
+      .chain(this.containerInfo)
+      .get('Config.Env')
+      .map(kv => _.slice(/^([^=]+)=(.*)$/.exec(kv), 1))
+      .fromPairs()
+      .value();
+  }
+
+  // getDetails() {
+  //   logger.info(`Building details hash for container '${this.containerName}'...`);
+  //   const info = {
+  //     'ID': this.containerInfo.Id,
+  //     'Name': this.containerInfo.Name,
+  //     'Created': utils.getTimeAgo(this.containerInfo.Created),
+  //     'Status': this.getContainerStatus(this.containerInfo.State)
+  //   };
+  //   const config = {
+  //     'Image': this.containerInfo.Config.Image,
+  //     'Entrypoint': _.join(this.containerInfo.Config.Entrypoint, ' '),
+  //     'Command': _.join(this.containerInfo.Config.Cmd, ' '),
+  //     'Work Directory': this.containerInfo.Config.WorkingDir,
+  //     'User': this.containerInfo.Config.User
+  //   };
+  //   const hostConfig = {
+  //     'CPU Shares': this.containerInfo.HostConfig.CpuShares,
+  //     'Memory': this.containerInfo.HostConfig.Memory,
+  //     'Memory Swap': this.containerInfo.HostConfig.MemorySwap,
+  //   };
+  //   const networkSettings = {
+  //     'Host IP': undefined,
+  //     'Container IP': this.containerInfo.NetworkSettings.IPAddress
+  //   };
+  //   const details = {
+  //     'Container': info,
+  //     'Configuration': config,
+  //     'Host Configuration': hostConfig,
+  //     'Environment Variables': this.getEnvironment()
+  //   };
+  //   if (this.containerInfo.State.Running) {
+  //     _.assign(hostConfig, {
+  //       'Privileged': this.containerInfo.HostConfig.Privileged,
+  //     });
+  //     const networkInfo = this.getNetworkInfo(this.containerInfo);
+  //     _.assign(networkSettings, {
+  //       'Host IP': networkInfo.ip
+  //     });
+  //     _.assign(details, {
+  //       'Network Settings': networkSettings,
+  //       'Exposed Ports': networkInfo.ports,
+  //       'Exposed Volumes': this.containerInfo.HostConfig.Binds
+  //     });
+  //   }
+  //   return details;
+  // }
+
+  static createDockerService(instanceId, options) {
+    const planId = options.plan_id;
+    const plan = catalog.getPlan(planId);
+    const context = _.get(options, 'context');
+    const dockerService = new DockerService(instanceId, plan);
+
+    return DockerImageLoaderService.load(plan)
+      .tap(manager => {
+        dockerService.imageInfo = manager.imageInfo;
+      })
+      .then(() => context ? context : dockerService.platformContext)
+      .then(context => dockerService.assignPlatformManager(DockerService.getPlatformManager(context.platform)))
+      .return(dockerService);
+  }
+
+  static getPlatformManager(platform) {
```

Comment:
this method should be removed. Any references to platformanagers should go out.

Reply:
Platform managers would still remain. SF2.0 managers are for the provisioning use case. Platform managers are for the consuming platform specific use case like SG creation. Cant be removed.